src/core/stdcpp/exception.d

```diff
@@ -6,95 +6,111 @@
  * Copyright: Copyright (c) 2016 D Language Foundation
  * License:   $(HTTP boost.org/LICENSE_1_0.txt, Boost License 1.0).
  * Authors:   $(HTTP digitalmars.com, Walter Bright)
+ *            Manu Evans
  * Source:    $(DRUNTIMESRC core/stdcpp/_exception.d)
  */
 
 module core.stdcpp.exception;
 
-extern (C++, "std"):
+import core.stdcpp.xutility : __cplusplus, CppStdRevision;
 
 version (CppRuntime_DigitalMars)
-{
-    import core.stdcpp.typeinfo;
+    version = GenericBaseException;
+version (CppRuntime_Gcc)
+    version = GenericBaseException;
+version (CppRuntime_Clang)
+    version = GenericBaseException;
 
-    alias void function() unexpected_handler;
-    unexpected_handler set_unexpected(unexpected_handler f) nothrow;
-    void unexpected();
-
-    alias void function() terminate_handler;
-    terminate_handler set_terminate(terminate_handler f) nothrow;
-    void terminate();
+extern (C++, "std"):
+@nogc:
 
-    bool uncaught_exception();
+///
+alias terminate_handler = void function() nothrow;
+///
+terminate_handler set_terminate(terminate_handler f) nothrow;
+///
+terminate_handler get_terminate() nothrow;
+///
+void terminate() nothrow;
 
-    class exception
-    {
-        this() nothrow { }
-        this(const exception) nothrow { }
-        //exception operator=(const exception) nothrow { return this; }
-        //virtual ~this() nothrow;
-        void dtor() { }
-        const(char)* what() const nothrow;
-    }
+static if (__cplusplus < CppStdRevision.cpp17)
+{
+    ///
+    alias unexpected_handler = void function();
+    ///
+    deprecated unexpected_handler set_unexpected(unexpected_handler f) nothrow;
+    ///
+    deprecated unexpected_handler get_unexpected() nothrow;
+    ///
+    deprecated void unexpected();
```

Comment:
they are.

Reply:
http://www.cplusplus.com/reference/exception/set_unexpected/ under Compatibility: 
>The use of dynamic-exception-specifiers is deprecated (since C++11).